Modules/Core/Common/include/itkFloodFilledSpatialFunctionConditionalConstIterator.hxx

```diff
@@ -112,7 +112,7 @@ FloodFilledSpatialFunctionConditionalConstIterator<TImage, TFunction>::IsPixelIn
       unsigned int counter;
       unsigned int counterCopy;
       unsigned int dim = TImage::ImageDimension;
-      auto         numReps = static_cast<unsigned int>(std::pow(static_cast<double>(2.0), static_cast<double>(dim)));
+      auto         numReps = static_cast<unsigned int>(std::pow(2.0, static_cast<double>(dim)));
```

Comment:
Thanks  @Leengit  I'm glad to see this `static_cast<double>(2.0)` being simplified to just `2.0` 😄   Note that with C++11/C++14, the other argument `static_cast<double>(dim)` should not need to be casted either, anymore, as the general form `pow(Arithmetic1 base, Arithmetic2 exp)` should be supported now with C++: https://en.cppreference.com/w/cpp/numeric/math/pow  However, we might as well defer further simplification of `std::pow` calls to another PR.

Reply:
Honestly I think it would be clearer to keep manual removal of redundant/unnecessary existing static_cast's separate from all those code changes that add static_cast's (in a separate commit, or even a separate PR). To just have a commit that only deals with old-style C-casts.

Reply:
@N-Dekker: Yes, I agree with both comments.  As far as changing this pull request to be consistent with them ... getting there from here would be a decent amount of work.  Is it good enough as is?

Reply:
@Leengit Didn't you run a script to do most of the job?

Reply:
Yes, a few `sed` commands.  Then fixed a few things by hand that my `sed` work got wrong.  Then more `sed` commands when I figured out whole categories that could be better.  Then more manual work to correct some more cases.

Reply:
The changes in this pull request from `T(x)` and `(T)x` are only for instances of `T` that are basic C++ types.  If there are cases that would benefit from similar changes, but where `T` is a class or template, those could be addressed in another pull request.

Reply:
I'm very interested to see those `sed` commands. Maybe I can use them myself in another project  😸

Reply:
Here they are in their ugliness as I used them.  Huge disclaimer: several of these did too little or too much and had to be corrected with subsequent commands.  And, in between many of them, there were edits by hand.  In short, these require careful handling and much work, and will still need fixes by hand.  You have been warned.  :smiley:

If you turn this into something less one-off, I'd be very interested to see it.

```sh
# It is important to use the top-level directory for the `git diff --name-only` commands
cd ~/git/ITK
# I should have used this alias more, from the start
alias findITKCXXFiles='(cd ~/git/ITK ; find * -type f | egrep -i -e '"'"'\.[it]?(cc|hh|[ch](\+\+|pp?|xx)?)$'"'"' | fgrep -v ThirdParty)'

egrep -lR '\((float|double|bool|char|short|int|long|unsigned| )+\)\(' | xargs sed -i -r -e 's/\((float|double|bool|char|short|int|long|unsigned| )+\)\(/static_cast<\1>(/g'
egrep -lR '\((float|double|bool|char|short|int|long|unsigned| )+\)\(' | xargs sed -i -r -e 's/\(((float|double|bool|char|short|int|long|unsigned| )+)\)\(/static_cast<\1>(/g'
findITKCXXFiles | xargs sed -i -r -e 's/double\(/static_cast<double>(/g'
git diff --name-only | xargs sed -i -r -e 's/rstatic_cast<double>/rdouble/g'
findITKCXXFiles | xargs sed -i -r -e 's/([^A-Za-z0-9_])float\(/\1static_cast<float>(/g'
findITKCXXFiles | xargs sed -i -r -e 's/([^A-Za-z0-9_])((float|double|bool|char|short|int|long|unsigned| )+)\(/\1static_cast<\2>(/g'
findITKCXXFiles | xargs sed -i -r -e 's/([^A-Za-z0-9_])((float|double|bool|char|short|int|long|unsigned)( (float|double|bool|char|short|int|long|unsigned))*)\(/\1static_cast<\2>(/g'
git diff --name-only | xargs sed -i -r -e 's/operator static_cast<bool>/operator bool/g'
git diff --name-only | xargs sed -i -r -e 's/return static_cast<bool>/return bool/g'
findITKCXXFiles | xargs sed -i -r -e 's/static_cast<float>\(([0-9\.]+)\)/\1f/g'
findITKCXXFiles | xargs sed -i -r -e 's/static_cast<float>\(([0-9]+)(\.([0-9]*))?\)/\1.\3f/g'
git diff --name-only | xargs sed -i -r -e 's/([0-9]+)\.f/\1.0f/g'
git diff master --name-only | xargs sed -i -r -e 's/std::function<static_cast<float>(float)>/std::function<float(float)>/g'
git diff master --name-only | xargs sed -i -r -e 's/std::function<static_cast<float>\(float\)>/std::function<float(float)>/g'
git diff master --name-only | xargs sed -i -r -e 's/std::function<static_cast<float>/std::function<float/g'
git diff master --name-only | xargs sed -i -r -e 's/std::function<static_cast<double>/std::function<double/g'
git diff master --name-only | xargs sed -i -r -e 's/std::function<static_cast<double>/std::function<double/g'
findITKCXXFiles | xargs sed -i -r -e  's/\(((float|double|bool|char|short|int|long|unsigned|const)( (float|double|bool|char|short|int|long|unsigned|const))*)\)((\[|\(|[->A-Za-z0-9_:])(\[|\]|\(|\)|[->\.A-Za-z0-9_:])*)/static_cast<\1>(\5)/g'
sed -i -r -e 's/static_cast < int\(*\) static_cast<int>\(> \(/static_cast<int (*)(int)>(/g'
sed -i -r -e 's/static_cast < int\(*\) static_cast<int>\(> \(/static_cast<int (*)(int)>(/g' Modules/IO/GDCM/src/itkGDCMImageIO.cxx
findITKCXXFiles | xargs sed -i -r -e 's/static_cast<double>\((-?[0-9]+\.[0-9]+([eE][0-9]+)?)\)/\1/g'
```